pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -462,6 +621,139 @@ pub mod pallet {
 
 			Ok((address, incoming_msg))
 		}
+
+		/// Iterates over the outbound messages stored in the queue and attempts
+		/// to process them.
+		///
+		/// If a message fails to process it is moved to the
+		/// `FailedOutboundMessages` storage so that it can be executed again
+		/// via the `process_failed_outbound_message` extrinsic.
+		fn service_outbound_message_queue(max_weight: Weight) -> Weight {
+			let mut weight_used = Weight::zero();
+
+			let mut processed_entries = Vec::new();
+
+			for (nonce, (domain, sender, message)) in OutboundMessageQueue::<T>::iter() {
+				processed_entries.push(nonce);
+
+				let mut weight = Weight::zero();
+
+				match Self::process_message(domain.clone(), sender.clone(), message.clone()) {
+					Ok(post_info) => {
+						Self::deposit_event(Event::OutboundMessageExecutionSuccess {
+							nonce,
+							sender,
+							domain,
+							message,
+						});
+
+						weight = weight
+							.saturating_add(T::DbWeight::get().writes(1))
+							.saturating_add(
+								post_info
+									.actual_weight
+									.expect("expected message processing weight"),
+							);
+					}
+					Err(e) => {
+						Self::deposit_event(Event::OutboundMessageExecutionFailure {
+							nonce,
+							sender: sender.clone(),
+							domain: domain.clone(),
+							message: message.clone(),
+							error: e.error,
+						});
+
+						FailedOutboundMessages::<T>::insert(
+							nonce,
+							(domain, sender, message, e.error),
+						);
+
+						weight = weight
+							.saturating_add(T::DbWeight::get().writes(2))
+							.saturating_add(
+								e.post_info
+									.actual_weight
+									.expect("expected message processing weight"),
+							);
+					}
+				}
+
+				weight_used = weight_used.saturating_add(weight);
```

Comment:
Nit: The match could directly return the processing `weight` such that we don't need to handle two mutable `weights`.

Reply:
Done.